server/src/index.js

```diff
@@ -27,30 +27,34 @@ const main = async () => {
       if (!await db.collection('Data').findOne({ url: data.url })) {
         const r = await db.collection('Data').insertOne(data)
         console.log('News added to the database.')
-        res.status(201).json({ success: true })
+        res.send(201)
       } else {
         console.log('Duplicate news found. Abort.')
-        res.status(409).json({ success: false })
+        res.send(409)
```

Comment:
You should return the existing ID so that the client can redirect the user to the page.